graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -254,14 +443,151 @@ def apply(self, input_df: DataFrame) -> DataFrame:
+            cols: list[str]
+                The list of columns the transformation is applied to. Order matters.
+
+            imputer_model: dict[str, Any]
+                A dict representation of the imputation applied.
+
+                Structure:
+                imputed_val_dict: dict[str, float]
+                    The imputed values for each column, {col_name: imputation_val}.
+                    Empty if no imputation was applied.
+                imputer_name: str
+                    The name of imputer used.
+
+            normalizer_model: dict[str, Any]
+                A dict representation of the normalization applied.
+
+                Structure:
+                norm_name: str
+                    The name of normalizer used.
+                norm_reconstruction: dict[str, Any]
+                    The reconstruction information for the normalizer. Empty if no normalization
+                    was applied. Inner structure depends on normalizer.
+
+                    Structure for MinMaxScaler:
+                    originalMinValues: list[float]
+                        The original minimum values for each column, in the order of the cols key.
+                    originalMaxValues: list[float]
+                        The original maximum values for each column, in the order of the cols key.
+
+                    Structure for StandardScaler:
+                    col_sums: dict[str, float]
+                        The sum of each column.
+
+            out_dtype: str
+                The output feature dtype, can take the values 'float32' and 'float64'.
+
+            transformation_name: str
+                Will be DistNumericalTransformation.
+        """
+        return self.json_representation
+
+    def apply_precomputed_transformation(self, input_df: DataFrame) -> DataFrame:
+        """Applies a numerical transformation using pre-computed representation.
+
+        Parameters
+        ----------
+        input_df : DataFrame
+            Input DataFrame to apply the transformation to.
+
+        Returns
+        -------
+        DataFrame
+            The input DataFrame, modified according to the pre-computed transformation values.
+        """
+        assert self.json_representation, (
+            "No precomputed transformation found. Please run `apply()` "
+            "first or set self.json_representation."
+        )
+
+        cols = self.json_representation["cols"]
+        impute_representation = self.json_representation["imputer_model"]
+        norm_representation = self.json_representation["normalizer_model"]
+        out_dtype = self.json_representation.get("out_dtype", TYPE_FLOAT32)
+
+        # First reapply pre-computed imputation if needed
+        if impute_representation["imputer_name"] == "none":
+            imputed_df = input_df
+        else:
+            imputed_vals = impute_representation["imputed_val_dict"]
+            shared_imputation = impute_representation["imputer_name"]
```

Comment:
Why use the name `shared_imputation`?

Reply:
We use this as opposed to having one imputation/norm per column which was in our initial design and we might still support in the future. Here all cols will share the same imputer and norm

Reply:
Can you add a comment here as "all cols will share the same imputer and norm"